velox/functions/prestosql/StringFunctions.h

```diff
@@ -430,4 +430,50 @@ struct LevenshteinDistanceFunction {
   }
 };
 
+template <typename T>
+struct NormalizeFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& string) {
+    auto form = "NFC";
+    doCall(result, string, form);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& string,
+      const arg_type<Varchar>& form) {
+    doCall(result, string, form);
+  }
+
+  FOLLY_ALWAYS_INLINE void doCall(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& string,
+      const arg_type<Varchar>& form) {
+    // Map for holding normalization form options
+    static const std::unordered_map<std::string, utf8proc_int16_t>
+        normalizatonMap{
+            {"NFC", (UTF8PROC_STABLE | UTF8PROC_COMPOSE)},
+            {"NFD", (UTF8PROC_STABLE | UTF8PROC_DECOMPOSE)},
+            {"NFKC", (UTF8PROC_STABLE | UTF8PROC_COMPOSE | UTF8PROC_COMPAT)},
+            {"NFKD", (UTF8PROC_STABLE | UTF8PROC_DECOMPOSE | UTF8PROC_COMPAT)}};
+
+    VELOX_USER_CHECK(
+        normalizatonMap.find(form) != normalizatonMap.end(),
+        "Normalization form must be one of [NFD, NFC, NFKD, NFKC]");
+    utf8proc_uint8_t* output;
+    auto outputLength = utf8proc_map(
```

Comment:
There probably isn't an alternative where this function lets you send a pre-allocated string and you can avoid the use of malloc.

Please add a comment indicating that this function newly allocates output using malloc so it should be free'd at the end.

Reply:
Comment missing here as requested?